checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -0,0 +1,233 @@
+        && rlTypeFactory.getDeclAnnotation(varElement, Owning.class) != null) {
+      checkOwningField(varElement);
+    }
+
+    return super.visitVariable(node, p);
+  }
+
+  /**
+   * Checks validity of a field {@code field} with an {@code @}{@link Owning} annotation. Say the
+   * type of {@code field} is {@code @MustCall("m"}}. This method checks that the enclosing class of
+   * {@code field} has a type {@code @MustCall("m2")} for some method {@code m2}, and that {@code
+   * m2} has an annotation {@code @EnsuresCalledMethods(value = "this.field", methods = "m")},
+   * guaranteeing that the {@code @MustCall} obligation of the field will be satisfied.
+   *
+   * @param field the declaration of the field to check
+   */
+  private void checkOwningField(Element field) {
+
+    if (checker.shouldSkipUses(field)) {
+      return;
+    }
+
+    // This value is side-effected.
+    List<String> unsatisfiedMustCallObligationsOfOwningField =
+        rlTypeFactory.getMustCallValue(field);
+
+    if (unsatisfiedMustCallObligationsOfOwningField.isEmpty()) {
+      return;
+    }
+
+    String error = "";
+    Element enclosingElement = field.getEnclosingElement();
+    List<String> enclosingMustCallValues = rlTypeFactory.getMustCallValue(enclosingElement);
+
+    if (enclosingMustCallValues == null) {
+      error =
+          " The enclosing element "
+              + ElementUtils.getQualifiedName(enclosingElement)
+              + " doesn't have a @MustCall annotation";
+    } else {
+      List<? extends Element> siblingsOfOwningField = enclosingElement.getEnclosedElements();
+      for (Element siblingElement : siblingsOfOwningField) {
+        if (siblingElement.getKind() == ElementKind.METHOD
+            && enclosingMustCallValues.contains(siblingElement.getSimpleName().toString())) {
+          AnnotationMirror ensuresCalledMethodsAnno =
+              rlTypeFactory.getDeclAnnotation(siblingElement, EnsuresCalledMethods.class);
+
+          if (ensuresCalledMethodsAnno != null) {
+            List<String> values =
+                AnnotationUtils.getElementValueArray(
+                    ensuresCalledMethodsAnno,
+                    rlTypeFactory.ensuresCalledMethodsValueElement,
+                    String.class);
+            for (String value : values) {
+              if (value.contains(field.getSimpleName().toString())) {
+                List<String> methods =
+                    AnnotationUtils.getElementValueArray(
+                        ensuresCalledMethodsAnno,
+                        rlTypeFactory.ensuresCalledMethodsMethodsElement,
+                        String.class);
+                unsatisfiedMustCallObligationsOfOwningField.removeAll(methods);
+              }
+            }
+            if (unsatisfiedMustCallObligationsOfOwningField.isEmpty()) {
+              return;
+            }
+          }
+
+          if (!unsatisfiedMustCallObligationsOfOwningField.isEmpty()) {
+            error =
```

Comment:
Minor: This variable may be reset multiple times.  It isn't used in the loop.  It could be set (if still null) when the error is reported, at the end of the method.

Reply:
I agree that that would be possible. However, I think the code is much more readable as-is, with the error message clearly set when the error is detected. This computation isn't that expensive (and only done in error cases at all, and only repeated if there are two of the same kind of error related to a single field), so I think it's fine as-is. I'll add a note, though.